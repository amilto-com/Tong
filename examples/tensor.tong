let l = import("linalg")
let a = l.ones([2,3])
let b = l.zeros([2,3])
let c = l.add(a,b)
print("shape:", l.shape(c))
print("rank:", l.rank(c))
print("(0,0):", l.get(c,[0,0]))
let d = l.tensor([1,2,3,4],[2,2])
let e = l.transpose(d)
let f = l.matmul(d,e)
print("matmul shape:", l.shape(f))
print(l.get(f,[0,0]), l.get(f,[0,1]), l.get(f,[1,0]), l.get(f,[1,1]))
