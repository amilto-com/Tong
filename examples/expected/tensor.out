shape: [2, 3] rank: 2 (0,0): 1.0 matmul shape: [2, 2] 5.0 11.0 11.0 25.0